syntax = "proto3";

package inference_server;

// Inference server
service Inference {
  // Inference
  rpc Inference(stream ListOfArrays) returns (stream ListOfArrays) {}
}

// Request and response
message ListOfArrays {
  repeated Array arrayList = 1;
}

// Represent an numpy array
message Array {
  string name = 1; // Node name in the neural network graph
  bytes raw_data = 2; // Binary numpy array in float32
  // Server should use the shape defined in the graph to reshape the array
}
